build_scripts/travis_script.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+"""A simple script to run Travis build steps."""
+
+from __future__ import print_function
+
+import collections
+import os
+import sys
+
+import build_utils
+
+def _ismod(f):
+  return f.endswith(".py") and not f.endswith("_test.py")
+
+
+STEP = collections.namedtuple("STEP", ["name", "command"])
+
+PYTYPE_COMMAND = os.path.join("out", "bin", "pytype")
+PYC_DIR = os.path.join(build_utils.PYTYPE_SRC_ROOT, "pytype", "pyc")
+PYC_FILES = [os.path.join(PYC_DIR, f) for f in os.listdir(PYC_DIR) if _ismod(f)]
+TYPECHECK_FILES = [
+    "pytype/compat.py", "pytype/debug.py", "pytype/utils.py"] + PYC_FILES
+
+
+def _begin_step(s):
+  print("")
+  print("BEGIN_STEP: %s" % s.name)
+  print("STEP_COMMAND: %s" % ' '.join(s.command))
+  print("")
+
+
+def _end_step(s):
+  print("\nEND_STEP: %s\n" % s.name)
+
+
+def _report_failure(s):
+  print("")
+  print(">>> STEP_FAILED: %s" % s.name)
+  print("")
+
+
+def _run_steps(steps):
+  for s in steps:
+    _begin_step(s)
+    returncode, _ = build_utils.run_cmd(s.command, pipe=False)
+    if returncode != 0:
+      _report_failure(s)
+      sys.exit(1)
+    _end_step(s)
+    
+
+def main():
+  s1 = STEP(name="Build",
+            command=["python", build_utils.build_script("build.py")])
+  s2 = STEP(name="Run Tests",
+            command=["python", build_utils.build_script("run_tests.py"), "-f"])
+  s3 = STEP(name="Type Check",
```

Comment:
I'd switch the RunTests and Type Check steps, since we think of type-checking as part of the build/compile step.

Reply:
I agree with that line of thought if we are using a "different" compiler. If we are using the in-tree compiler, we should first ensure that the compiler works before we use it. That is why I changed the order.

Reply:
Ah, that makes sense. I don't like that the steps feel out of order, but it does seem like the most sensible thing to do.